### DISCUSSION TITLE
Does Blacksheep support built-in GZip compression ?

### DISCUSSION THREAD

**Discussion by i-Ching** [Neoteroi/BlackSheep#293] [Q&A] [answered]
- Starlette includes GZipMiddleware: https://www.starlette.io/middleware/
- Starlite includes Compression middleware: https://starlite-api.github.io/starlite/usage/7-middleware/3-builtin-middlewares/4-compression-middleware/

Does Blacksheep support built-in GZip compression ?

**Comment by RobertoPrevato** [comment, marked as answer]
Hi @i-Ching 
No, BlackSheep does not support built-in GZip compression at the moment. It's planned for the future.

There is not a goal to have feature parity with Starlette, and comparing the two frameworks is a bit pointless. BlackSheep also supports some built-in features that are not built-in supported in Starlette, just to mention three:

- ~dependency injection~ [UPDATE] I see Starlette finally added support for dependency injection, but it doesn't look as mature as in BlackSheep, since it requires describing dependencies on routes and boilerplate that is not required in blacksheep.
- support for HTTP Range requests inside the framework
- built-in support for OpenID Connect
